### DISCUSSION TITLE
Use @Encoded with "RESTEasy REST Client"

### DISCUSSION THREAD

**Discussion by sraturi** [quarkusio/quarkus#23935] [Q&A]
Hello,

I am following this guide https://quarkus.io/guides/rest-client to communicate with a 3rd party API. Overall everything works but there is a small issue I am running with query param.

I am able to set up query param as described in the tutorial. However, I am not able to use the `Encoded` string as a query param.

For example

I have this endpoint
 ```

@Path("/device")
    @GET
    fun deviceSelect(@QueryParam("\$select")  selectQuery: String): ListOf<DeviceMeta>
```
Also, the third-party follows the [odata](https://www.odata.org/getting-started/basic-tutorial/#select) conventions.

Now when I pass in a single field into this method, everything works, I get a list with that field.

However, when I put in multiple fields, so something like `deviceSelect("name,id")`, I run into malformed URL. Basically, the other endpoint is expecting the actual `,` but our endpoint is passing it with % encoded.

i.e valid: `/device?$select=name,id`
whats generated: `/device?%24select=name%2Cid` 

I know the third-party API is supposed to handle the encoded inputs but I need a solution while they fix that.

I tried annotating the param, method, and class with `@Encoded` but had no luck...

[a comment by an automated account is not shown]

**Comment by michalszynkiewicz** [comment]
We try to put more focus on REST Client Reactive now. Can you check if it works there?

**Reply by sraturi** [reply]
yes, I had the same issue, I modified the `rest-client-reactive-quickstart` guide

**Reply by sraturi** [reply]
To be clear, I added encoded annotation to the method and/or to the query parameter but special characters like `,` and `/` still got encoded with `%`

**Reply by michalszynkiewicz** [reply]
also, have you tried it with java?

**Reply by michalszynkiewicz** [reply]
I created https://github.com/quarkusio/quarkus/issues/23961 for it, it looks like this is unsuppoted at the moment
